app/models/manageiq/providers/vmware/infra_manager/refresh_parser/filter.rb

```diff
@@ -32,7 +32,7 @@ def filter_vc_data(ems, target)
         vm_data = vm_inv_by_vm(target)
         unless vm_data.nil?
           filtered_data[:vm] = vm_data
-          filtered_data[:storage] = storage_inv_by_vm_inv(vm_data)
+          filtered_data[:storage] = storage_inv_by_host_inv(host_inv_by_host(target.host))
```

Comment:
Talked with @jameswnl and I didn't realize this was the filter code...So, this technique seems fine, but I still don't think it should use target.host...instead it should rely on the host as presented by the vm_data.

Also, we need a test to verify that since other storages are coming in, that those don't sever the connection to untargeted VMs.